rust/sdk-processor/src/steps/parquet_default_processor/parquet_default_extractor.rs

```diff
@@ -0,0 +1,146 @@
+pub struct ParquetDefaultExtractor
+where
+    Self: Processable + Send + Sized + 'static,
+{
+    pub opt_in_tables: Option<Vec<String>>,
+}
+
+type ParquetTypeMap = HashMap<ParquetTypeEnum, ParquetTypeStructs>;
+
+#[async_trait]
+impl Processable for ParquetDefaultExtractor {
+    type Input = Vec<Transaction>;
+    type Output = ParquetTypeMap;
+    type RunType = AsyncRunType;
+
+    async fn process(
+        &mut self,
+        transactions: TransactionContext<Self::Input>,
+    ) -> anyhow::Result<Option<TransactionContext<ParquetTypeMap>>, ProcessorError> {
+        let (move_resources, write_set_changes, parquet_transactions, table_items, move_modules) =
+            process_transactions(transactions.data);
+
+        // Print the size of each extracted data type
+        debug!("Processed data sizes:");
+        debug!(" - MoveResources: {}", move_resources.len());
+        debug!(" - WriteSetChanges: {}", write_set_changes.len());
+        debug!(" - ParquetTransactions: {}", parquet_transactions.len());
+        debug!(" - TableItems: {}", table_items.len());
+        debug!(" - MoveModules: {}", move_modules.len());
+
+        let mut map: HashMap<ParquetTypeEnum, ParquetTypeStructs> = HashMap::new();
+
+        // Array of tuples for each data type and its corresponding enum variant
+        let data_types = [
+            (
+                ParquetTypeEnum::MoveResource,
+                ParquetTypeStructs::MoveResource(move_resources),
+            ),
+            (
+                ParquetTypeEnum::WriteSetChange,
+                ParquetTypeStructs::WriteSetChange(write_set_changes),
+            ),
+            (
+                ParquetTypeEnum::Transaction,
+                ParquetTypeStructs::Transaction(parquet_transactions),
+            ),
+            (
+                ParquetTypeEnum::TableItem,
+                ParquetTypeStructs::TableItem(table_items),
+            ),
+            (
+                ParquetTypeEnum::MoveModule,
+                ParquetTypeStructs::MoveModule(move_modules),
+            ),
+        ];
+
+        // Populate the map based on opt-in tables
+        for (enum_type, data) in data_types {
+            add_to_map_if_opted_in_for_backfill(&self.opt_in_tables, &mut map, enum_type, data);
+        }
+
+        debug!(
+            "Map populated with data for the following tables: {:?}",
+            map.keys().collect::<Vec<_>>()
+        );
+
+        Ok(Some(TransactionContext {
+            data: map,
+            metadata: transactions.metadata,
+        }))
+    }
+}
+
+pub fn process_transactions(
+    transactions: Vec<Transaction>,
+) -> (
+    Vec<MoveResource>,
+    Vec<WriteSetChangeModel>,
+    Vec<TransactionModel>,
+    Vec<TableItem>,
+    Vec<MoveModule>,
+) {
+    // this will be removed in the future.
+    let mut transaction_version_to_struct_count = AHashMap::new();
+    let (txns, _, write_set_changes, wsc_details) = TransactionModel::from_transactions(
```

Comment:
why is the second value dropped?

Reply:
it's because it's returning the metadata that we don't use here